### PR TITLE
Add read methods

### PR DESCRIPTION
**Pull request by germartinez**
Add read methods to access the data from the Safe contract

### CODE REVIEW

File: src/EthersSafe.ts [hunk 1]
```diff
@@ -10,31 +10,89 @@ class EthersSafe implements Safe {
   #ethers: any
   #signer: Wallet
 
+  /**
+   * Creates an instance of the Safe Core SDK.
+   *
+   * @param ethers - Ethers v5 library
+   * @param signer - Ethers signer
+   * @param safeAddress - The address of the Safe account to use
+   * @returns The Safe Core SDK instance
```
Comment (mikheevm) [inline src/EthersSafe.ts:19]: shouldn't return also include a type? https://jsdoc.app/tags-returns.html

btw I think typescript supports jsdoc and can generate types from it, so maybe we don't need to duplicate them: https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html#param-and-returns

File: README.md [hunk 2]
```diff
@@ -1 +1,81 @@
...
+1. Create a Safe account with two owners and threshold equal two (this configuration is just an example):
+```js
+import { ethers } from 'ethers')
+import EthersSafe, { makeSafeTransaction } from 'safe-core-sdk'
+
+const provider = ethers.getDefaultProvider('homestead')
+const wallet1 = ethers.Wallet.createRandom().connect(provider)
+const wallet2 = ethers.Wallet.createRandom().connect(provider)
+
+const safe = await safeContract.setup(
```
Comment (rmeissner) [inline README.md:38]: Where is the `safeContract` coming from?

File: src/EthersSafe.ts [hunk 3]
```diff
@@ -52,6 +110,24 @@ class EthersSafe implements Safe {
     return txHash
   }
 
+  /**
+   * Signs data using the current owner account.
+   *
+   * @param hash - The data to sign
+   * @returns The Safe signature
+   */
+  async signMessage(hash: string): Promise<SafeSignature> {
```
Comment (rmeissner) [inline src/EthersSafe.ts:119]: I am not sure about the name as the Safe also has a method called `signMessage`

File: src/utils/transactions.ts [hunk 4]
```diff
@@ -49,6 +49,12 @@ export class SafeTransaction {
   }
 }
 
+/**
```
Comment (rmeissner) [inline src/utils/transactions.ts:52]: the baseOfsset in line 40 should probably be `signers.length * 65` as we need the offset in bytes not in chars.

**Review by rmeissner — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (germartinez) [reply README.md:38]: How would you represent an existing Safe account here? It doesn't make sense to add all the logic to explain where this is coming from.

Comment (germartinez) [reply README.md:38]: It is also weird that a direct interaction with the contract is needed before using the SDK. I guess at some point the SDK should allow to create Safes, right? Should this be prioritized?

File: src/EthersSafe.ts (hunk 3 above)
Comment (germartinez) [reply src/EthersSafe.ts:119]: Is `signTransactionHash` better?

File: README.md (hunk 2 above)
Comment (rmeissner) [reply README.md:38]: maybe just add a comment like
```js
// Existing Safe address (e.g. create from a Safe created via https://app.gnosis-safe.io
const safeAddress = "0x<some_safe_address>"
```

File: src/EthersSafe.ts (hunk 3 above)
Comment (rmeissner) [reply src/EthersSafe.ts:119]: yes, this would make it a lot clearer for me :)

**Pull request merged**